File: DungeonDebt/Assets/Scripts/Data/HeroDefinition.cs
public class HeroDefinition
{
    public HeroDefinition(
        string id,
        string displayName,
        HeroRole role,
        int baseAttack,
        int baseHealth,
        int baseUpkeep,
        string effectDescription,
        HeroEffectId effectId)
    {
        Id = id;
        DisplayName = displayName;
        Role = role;
        BaseAttack = baseAttack;
        BaseHealth = baseHealth;
        BaseUpkeep = baseUpkeep;
        EffectDescription = effectDescription;
        EffectId = effectId;
    }

    public string Id { get; }
    public string DisplayName { get; }
    public HeroRole Role { get; }
    public int BaseAttack { get; }
    public int BaseHealth { get; }
    public int BaseUpkeep { get; }
    public string EffectDescription { get; }
    public HeroEffectId EffectId { get; }
}

File: web/src/data/HeroDefinition.js
// Ported from DungeonDebt/Assets/Scripts/Data/HeroDefinition.cs
// Immutable hero template.
export class HeroDefinition {
  constructor(id, displayName, role, baseAttack, baseHealth, baseUpkeep, effectDescription, effectId,
    passiveAbility = null, activeAbility = null) {
    this.id = id;
    this.displayName = displayName;
    this.role = role;
    this.baseAttack = baseAttack;
    this.baseHealth = baseHealth;
    this.baseUpkeep = baseUpkeep;
    this.effectDescription = effectDescription;
    this.effectId = effectId;
    this.passiveAbility = passiveAbility;
    this.activeAbility = activeAbility;
    Object.freeze(this);
  }
}
